graphql_compiler/cost_estimation/int_value_conversion.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from uuid import UUID
+import datetime
+
+from .helpers import is_int_field_type, is_uuid4_type, is_datetime_field_type, is_date_field_type
+
+
+# UUIDs are defined in RFC-4122 as a 128-bit identifier. This means that the minimum UUID value
+# (represented as a natural number) is 0, and the maximal value is 2^128-1.
+MIN_UUID_INT = 0
+MAX_UUID_INT = 2**128 - 1
+
+
+# This module is a utility for reasoning about intervals when computing filter selectivities,
+# and generating parameters for pagination. Since integers are the easiest type to deal with
+# in this context, when we encounter a different type we represent it as an int, do all the
+# computation in the integer domain, and transfer the computation back into the original domain.
+#
+# In order to be able to reason about value intervals and successor/predecessor values, we
+# make sure these mappings to integers are increasing bijective functions.
+#
+# This kind of mapping is easy to do for int, uuid and datetime types, but not possible for other
+# types, like string. When the need for other types arises, the precise interface for range
+# reasoning can be defined and implemented separately for each type.
```

Comment:
This is what we can do whenever we need more generality from this: https://github.com/kensho-technologies/graphql-compiler/blob/c54e165b32e461c3c6947a9e5390c1697c4f73e1/graphql_compiler/cost_estimation/filter_set.py